ilastik/workflows/tracking/common.py

```diff
@@ -0,0 +1,48 @@
+###############################################################################
+#   ilastik: interactive learning and segmentation toolkit
+#
+#       Copyright (C) 2011-2020, the ilastik developers
+#                                <team@ilastik.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# In addition, as a special exception, the copyright holders of
+# ilastik give you permission to combine ilastik with applets,
+# workflows and plugins which are not covered under the GNU
+# General Public License.
+#
+# See the LICENSE file for details. License information is also available
+# on the ilastik web site at:
+#          http://ilastik.org/license.html
+###############################################################################
+from dataclasses import dataclass
+from typing import SupportsInt
+
+
+@dataclass(frozen=True)
+class PluralInflector:
+    zero: str
+    one: str
+    many: str
+
+    # arbitrarily defined maximum in case list() is called on this object
+    max_index: int = 256
```

Comment:
If we decided to be general, might as well generalize this too:

```python
def inflect(n, *, zero, one, many):
    if n < 0:
        raise ValueError(f"cannot inflect negative number {n}")
    if n == 0:
        return zero
    if n == 1:
        return self.one
    return self.many.format(n)


class LazySeq(collection.abc.Sequence):
    def __init__(self, func, *, size):
        self.func = func
        self.size = size

    def __getitem__(self, i):
        if not (0 <= i < self.size):
            raise IndexError(i)
        return self.func(i)

    def __len__(self):
        return self.size


DIVISION_CLASSIFIER_LABEL_NAMES = LazySeq(
    functools.partial(inflect, zero="False Detection", one="1 Object", many="{} Objects"), size=256
)

Reply:
okay, thanks for the thorough review on that. I'll go with the current solution.